scripts/start/readCheckScenarioConfig.R

```diff
@@ -82,20 +82,10 @@ readCheckScenarioConfig <- function(filename, remindPath = ".", testmode = FALSE
     message(msg)
   }
   if ("path_gdx_bau" %in% names(scenConf)) {
-    # fix if bau given despite not needed
-    needBau <- list(welfare = "ineqLognormal",
-                    carbonprice = c("NDC", "diffPriceSameCost"),
-                    carbonpriceRegi = "NDC",
-                    emicapregi = "AbilityToPay")
+    # fix if bau given despite not needed. needBau is defined in needBau.R
     scenNeedsBau <- rep(FALSE, nrow(scenConf))
-    for (n in names(needBau)) {
-      if (n %in% names(scenConf)) {
-        if (n == "welfare" && "cm_emiscen" %in% names(scenConf)) {
-          scenNeedsBAU <- scenNeedsBau | (scenConf[[n]] %in% needBau[[n]] & scenConf[["cm_emiscen"]] != 1)
-        } else {
-          scenNeedsBau <- scenNeedsBau | scenConf[[n]] %in% needBau[[n]]
-        }
-      }
+    for (n in intersect(names(needBau), names(scenConf))) {
+      scenNeedsBau <- scenNeedsBau | scenConf[[n]] %in% needBau[[n]]
```

Comment:
In L86 `scenNeedsBau` has been defined as a vector. Still needed?

Reply:
Yes, it is still a vector, as `scenConf[[n]]` is also vector (for example, the `carbonprice` column). So `scenNeedsBau` starts as being `FALSE` everywhere, and once any of these columns has an element that needs a `path_gdx_bau`, it is turned to `TRUE` but only in the entry corresponding to that row.

Reply:
Oh sorry, I didn't notice, that `scenConf[[n]]` is also vector.